uaclient/system.py

```diff
@@ -34,7 +34,7 @@
     r"[.-]"
     r"(?P<subrev>[\d]+)"
     r"(~(?P<hwerev>[\d.]+))?"
-    r"-"
+    r"[-+]"
```

Comment:
I think you need to escape the "+" right? Won't this catch "one or more dashes"?

Reply:
TIL it does not because of the brackets